[Button Layout] `buttons` should not have `align-items: flex-start` by default

Ahmad-S792 opened the pull request:
#### d12e44816fa89ab09bf67649ec7178fffabb8b71
<pre>
[Button Layout] `buttons` should not have `align-items: flex-start` by default

<a href="https://bugs.webkit.org/show_bug.cgi?id=289441">https://bugs.webkit.org/show_bug.cgi?id=289441</a>
<a href="https://rdar.apple.com/146615626">rdar://146615626</a>

Reviewed by Tim Nguyen.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

Merge: <a href="https://chromium-review.googlesource.com/c/chromium/src/+/6013405">https://chromium-review.googlesource.com/c/chromium/src/+/6013405</a>

This patch removes unneeded `align-items: flex-start`, since it is leftover
from our implementation of `button` as flex containers in 228400@main.

* Source/WebCore/css/html.css:
(input:is([type=&quot;button&quot;], [type=&quot;submit&quot;], [type=&quot;reset&quot;]), input[type=&quot;file&quot;]::file-selector-button, button):
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/widgets/button-layout/flex-expected.txt: Rebaselined
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/rendering/widgets/button-layout/flex-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/rendering/widgets/button-layout/flex-expected.txt: Removed.

Canonical link: <a href="https://commits.webkit.org/291883@main">https://commits.webkit.org/291883@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f553e7eaaef12659f6b9e62d9fa40f900074a572

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94306 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13893 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3664 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99318 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44834 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/96356 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14193 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22323 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71949 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29288 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97308 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10545 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/85157 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52294 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10234 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/2840 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44152 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80466 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2932 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101363 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21358 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/15577 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80954 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21610 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81170 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80336 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20023 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24886 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2261 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/14580 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21342 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/26521 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21029 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/24489 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22770 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (approved):
Can you remove this too: https://searchfox.org/wubkat/rev/662cdccd8338a60684dac2fc6e14d26305e5db8a/Source/WebCore/css/html.css#1127 ?

The select children can't really be laid out as a flex items as it stands, so that rule does nothing.

nt1m commented on Source/WebCore/css/html.css:
```diff
@@ -1124,7 +1123,6 @@ select {
 
 #if !(defined(WTF_PLATFORM_IOS_FAMILY) && WTF_PLATFORM_IOS_FAMILY)
 select:is([size], [multiple]), select[size][multiple] {
-    align-items: flex-start;
```
Can you leave that unchanged for this bug? We should remove it, but also remove `align-items: center` to avoid getting incorrect results.

Ahmad-S792 replied:
Can do. :-)

Ahmad-S792 replied:
Done.

Ahmad-S792 committed:
[Button Layout] `buttons` should not have `align-items: flex-start` by default

https://bugs.webkit.org/show_bug.cgi?id=289441
rdar://146615626

Reviewed by Tim Nguyen.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

Merge: https://chromium-review.googlesource.com/c/chromium/src/+/6013405

This patch removes unneeded `align-items: flex-start`, since it is leftover
from our implementation of `button` as flex containers in 228400@main.

* Source/WebCore/css/html.css:
(input:is([type="button"], [type="submit"], [type="reset"]), input[type="file"]::file-selector-button, button):
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/widgets/button-layout/flex-expected.txt: Rebaselined
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/rendering/widgets/button-layout/flex-expected.txt: Removed.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/rendering/widgets/button-layout/flex-expected.txt: Removed.

Canonical link: https://commits.webkit.org/291883@main
